{
  bestGetCheckedTypeValidator().validateClass(exceptionClass);
  try {
    return future.get(timeout,unit);
  }
 catch (  InterruptedException e) {
    currentThread().interrupt();
    throw newWithCause(exceptionClass,e);
  }
catch (  TimeoutException e) {
    throw newWithCause(exceptionClass,e);
  }
catch (  ExecutionException e) {
    wrapAndThrowExceptionOrError(e.getCause(),exceptionClass);
    throw new AssertionError();
  }
}
